### PR TITLE
dune_2: 2.7.0 -> 2.7.1

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change
https://github.com/ocaml/dune/releases/tag/2.7.1

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg build flow ocamlformat beluga satysfi fstar

File: pkgs/development/tools/ocaml/dune/2.nix [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
     url = "https://github.com/ocaml/dune/releases/download/${version}/dune-${version}.tbz";
-    sha256 = "058wiyncczbmlfxj3cnwn5n68wkmbaf4mgjm2bkp2hffpn2wl5xl";
+    sha256 = "0lnfmc06yhbdiv6mmp0aksnnvzz4aw4zabrdg89p5msyzir3qrvn";
   };
 
   buildInputs = [ ocaml findlib ];
 
   buildFlags = "release";
 
+  patches = [
```
Comment (vbgl) [inline pkgs/development/tools/ocaml/dune/2.nix:20]: This attribute should be inherited in dune-build-info…

**Review by vbgl — CHANGES_REQUESTED** [review]
dune-configurator now depends on [csexp](https://github.com/ocaml-dune/csexp)

**Comment by marsam** [comment]
@GrahamcOfBorg build flow ocamlformat beluga satysfi fstar

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.irmin_1

**Comment by vbgl** [comment]
Updating `result` breaks `irmin_1`:

> File "src/irmin/type.ml", line 1123, characters 21-24:
1123 |   let rec t: type a. a t -> a decode = function
                            ^^^
Error: The type constructor t expects 2 argument(s),
       but is here applied to 1 argument(s)

**Comment by vbgl** [comment]
There is an easy patch to csexp:

~~~
diff --git a/pkgs/development/ocaml-modules/csexp/default.nix b/pkgs/development/ocaml-modules/csexp/default.nix
index 826372ef196..51274890700 100644
--- a/pkgs/development/ocaml-modules/csexp/default.nix
+++ b/pkgs/development/ocaml-modules/csexp/default.nix
@@ -1,4 +1,4 @@
-{ lib, fetchurl, buildDunePackage, result }:
+{ lib, fetchurl, buildDunePackage }:
 
 buildDunePackage rec {
   pname = "csexp";
@@ -6,14 +6,16 @@ buildDunePackage rec {
 
   useDune2 = true;
 
-  minimumOCamlVersion = "4.02.3";
+  minimumOCamlVersion = "4.08";
 
   src = fetchurl {
     url = "https://github.com/ocaml-dune/csexp/releases/download/${version}/csexp-${version}.tbz";
     sha256 = "0maihbqbqq9bwr0r1cv51r3m4hrkx9cf5wnxcz7rjgn13lcc9s49";
   };
 
-  propagatedBuildInputs = [ result ];
+  postPatch = ''
+    substituteInPlace src/csexp.ml --replace Result.result Result.t
+  '';
 
   meta = with lib; {
     homepage = "https://github.com/ocaml-dune/csexp";
~~~

**Comment by marsam** [comment]
should we still maintain `irmin_1`? 
AFAIK all the upstream development is going to irmin 2.x; https://github.com/NixOS/nixpkgs/pull/74818 mentions that it was introduced for tezos (not yet in nixpkgs), and tezos [already uses irmin 2](https://gitlab.com/tezos/tezos/-/merge_requests/1956) cc: @alexfmpe irmin_1 maintainer

**Comment by vbgl** [comment]
IMHO, updating `result`, removing `irmin_1`, and updating `dune_2` are three independent issues. Testing an update of dune is painful enough. Please let’s focus on `dune_2` in this PR.

**Comment by marsam** [comment]
I agree. I applied your patch, thanks!

@GrahamcOfBorg build flow ocamlformat beluga satysfi fstar

**Review by vbgl — APPROVED** [review]

**Pull request merged**